zipline/sources/benchmark_source.py

```diff
@@ -46,11 +48,10 @@ def __init__(self, benchmark_sid, env, trading_calendar, sessions,
                     self.sessions,
                     self.data_portal
                 )
-        else:
+        elif benchmark_returns is not None:
             # get benchmark info from trading environment, which defaults to
             # downloading data from Yahoo.
-            daily_series = \
-                env.benchmark_returns[sessions[0]:sessions[-1]]
+            daily_series = benchmark_returns[sessions[0]:sessions[-1]]
```

Comment:
Good catch. I'll move that to where the returns are retrieved in `algorithm.py`.